tests/test_requests.py

```diff
@@ -1577,6 +1577,16 @@ def test_auth_is_stripped_on_redirect_off_host(self, httpbin):
         assert r.history[0].request.headers['Authorization']
         assert not r.request.headers.get('Authorization', '')
 
+    def test_auth_is_stripped_on_scheme_redirect(self, httpbin, httpbin_secure, httpbin_ca_bundle):
+        r = requests.get(
+            httpbin_secure('redirect-to'),
+            params={'url': httpbin('get')},
+            auth=('user', 'pass'),
+            verify=httpbin_ca_bundle
+        )
+        assert r.history[0].request.headers['Authorization']
+        assert not r.request.headers.get('Authorization', '')
```

Comment:
We should be explicit here and validate that `Authorization` is not in the request headers rather than checking if the value is falsey.

Reply:
I was just copying the test above it, but I agree that would be an improvement. Will fix it tomorrow.